src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6476,6 +6476,24 @@ private boolean nextPacket() throws SQLServerException {
         payloadOffset = 0;
         return true;
     }
+    
+    boolean hasNextPacket() throws SQLServerException {
+        if (null == currentPacket.next) {
+            return false;
+        }
+        else {
+            return true;
+        }
+    }
+
+    boolean endOfCurrentPacket() {
+        if (payloadOffset == currentPacket.payloadLength) {
+            return true;
+        }
+        else {
+            return false;
+        }
```

Comment:
Method `endOfCurrentPacket()` content can be compressed to:
`return (payload == currentPacket.payloadLength);`